src/electron/modules/recently-played.ts

```diff
@@ -39,45 +39,38 @@ export class RecentlyPlayedModule extends WebsocketModule {
       const summoners = json.map((summoner: RecentlyPlayedSummoner) => new RecentlyPlayedSummoner(summoner));
 
       // sort newest first
-      const orderedSummoners = summoners.sort((summonerA: RecentlyPlayedSummoner, summonerB: RecentlyPlayedSummoner) => summonerB.gameCreationDate.diff(summonerA.gameCreationDate));
+      const sortedSummoners = summoners.sort((summonerA: RecentlyPlayedSummoner, summonerB: RecentlyPlayedSummoner) => summonerB.gameCreationDate.diff(summonerA.gameCreationDate));
 
       // keep newest duplicates
-      const uniqueSummoners = orderedSummoners.filter((summoner: RecentlyPlayedSummoner, index: number) => {
-        return orderedSummoners.findIndex((newestSummoner: RecentlyPlayedSummoner) => newestSummoner.summonerId === summoner.summonerId) === index;
+      const uniqueSummoners = sortedSummoners.filter((summoner: RecentlyPlayedSummoner, index: number) => {
+        return sortedSummoners.findIndex((newestSummoner: RecentlyPlayedSummoner) => newestSummoner.summonerId === summoner.summonerId) === index;
       }).slice(0, this.recentSummonerLimit);
 
-      // create object by game id
-      const summonersByGame = uniqueSummoners.reduce((acc, post) => {
-        const gameId = post.gameId;
-        if (gameId in acc) {
-          acc[gameId].push(post);
-        } else {
-          acc[gameId] = [post];
-        }
-        return { ...acc };
+      // group summoners by game
+      const summonersByGame = uniqueSummoners.reduce((summonersByGame: Record<number, RecentlyPlayedSummoner[]>, summoner: RecentlyPlayedSummoner) => {
```

Comment:
Can you explain why the accumulator is a Record rather than just an array of RecentlyPlayedSummoners?

Reply:
The objective is to group the players in each game together. In other words, create a mapping from the game ID to the array of players with that game ID. This can be done by reducing where the accumulator is the mapping. Each step of the reduction updates the mapping by adding the current player to the appropriate array within the mapping.

Reply:
I was unfamiliar with Record types. Makes sense now, thanks!

Reply:
Record is a TypeScript type (i.e. there is no JavaScript Record object/class). It’s essentially what you use when you have a generic object being used for a map (i.e. `{“a”: 1, “b”: 2, “c”: 3}` would have type `Record<string, number>`). See https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype.